src/UserSpaceInstrumentation/include/UserSpaceInstrumentation/Attach.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef USER_SPACE_INSTRUMENTATION_ATTACH_H_
+#define USER_SPACE_INSTRUMENTATION_ATTACH_H_
+
+#include <sys/types.h>
+
+#include "OrbitBase/Result.h"
+
+namespace orbit_user_space_instrumentation {
+
+[[nodiscard]] ErrorMessageOr<void> AttachAndStopProcess(pid_t pid);
+
+[[nodiscard]] ErrorMessageOr<void> DetachAndContinueProcess(pid_t pid);
```

Comment:
(This looks very much like one of those cases where `[[nodiscard]]` is added because we add it everywhere but where the caller might want to ignore the result. Not that it matters, but I'm still on the fence with out approach to nodiscard.)

Reply:
Hmm, I think enforcing checking for errors here is valid - the underlying calls to ptrace can fail in unusual cases.